### PR TITLE
chore: improve zigbee2mqtt maintenance path

### PR DESCRIPTION
**Pull request by quyentonndbs**
Summary:
- Add or tighten focused edge-case tests or type assertions in index.js, test/data.test.ts related to TypeScript, frontend tooling, test stability, build fixes; avoid docs-only changes and broad refactors.
- Keep the change narrow so it is straightforward to review.

Notes:
- I kept this scoped to the relevant implementation and tests.

### CODE REVIEW

File: test/data.test.ts [hunk 1]
```diff
@@ -15,12 +15,21 @@ describe("Data", () => {
...
             process.env.ZIGBEE2MQTT_DATA = expected;
             data._testReload();
-            const actual = data.getPath();
-            expect(actual).toBe(expected);
-            expect(data.joinPath("test")).toStrictEqual(path.join(expected, "test"));
-            expect(data.joinPath("/test")).toStrictEqual(path.resolve(expected, "/test"));
-            delete process.env.ZIGBEE2MQTT_DATA;
-            data._testReload();
+
+            try {
```
Comment (Koenkk) [inline test/data.test.ts:19]: What's the goal of wrapping this in a try/catch?

Comment (Nerivec) [reply test/data.test.ts:19]: try/finally
It always deletes the env and reloads (doesn't affect other tests if this one ever fails).

Really we should be moving all tests to use
https://vitest.dev/api/vi.html#vi-stubenv
with (in `afterEach`)
https://vitest.dev/api/vi.html#vi-unstuballenvs

**Comment by Koenkk** [comment]
thanks!
